src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -66,6 +68,8 @@ public static int Main(string[] args)
                 var outputValue = output.Value();
                 var intermediateValue = intermediateOutput.Value();
 
+                var preconditions = new Preconditions();
```

Comment:
Very good question. I am now reworking all of this into a separate build command that does the checking.

With centralized checking I am afraid that Command.create might have weird process and invocation dir implicit dependencies on Directory.GetCurrentDirectory or on AppContext.BaseString. If that's so, Command.Create might return different things depending on the executing process and the invocation directory. This means I would not get the ground truth unless I either actually collect path probing preconditions just in time, or try to infer and simulate the right paths as if the command were executed. I'll do some hello world experiments to see how horrible this gets.